govc/vm/clone.go

```diff
@@ -316,6 +320,49 @@ func (cmd *clone) cloneVM(ctx context.Context) (*object.Task, error) {
 		Template: cmd.template,
 	}
 
+	if cmd.link == "snapshot" {
+		relocateSpec.DiskMoveType = string(types.VirtualMachineRelocateDiskMoveOptionsCreateNewChildDiskBacking)
+		cloneSpec.Location = relocateSpec
+		//get snapshots:
+		var o mo.VirtualMachine
+		vm := cmd.VirtualMachine
+		err = vm.Properties(ctx, vm.Reference(), []string{"snapshot"}, &o)
```

Comment:
Thanks @amitbet , I think we could simplify this further by changing `FindSnapshot` to return CurrentSnapshot if name == "" (diff below).  Then we don't need to load the snapshot property here and validation/errors are all handled within `FindSnapshot`.  What do you think?

```diff
modified   object/virtual_machine.go
@@ -534,6 +534,10 @@ func (v VirtualMachine) FindSnapshot(ctx context.Context, name string) (Referenc
 		return nil, errors.New("No snapshots for this VM")
 	}
 
+	if name == "" {
+		return o.Snapshot.CurrentSnapshot, nil
+	}
+
 	m := make(snapshotMap)
 	m.add("", o.Snapshot.RootSnapshotList)
```